src/file.rs

```diff
@@ -43,11 +77,66 @@ impl HasPath for PubkyAppFile {
 }
 
 impl Validatable for PubkyAppFile {
-    // TODO: content_type validation.
+    fn sanitize(self) -> Self {
+        let name = self.name.trim().chars().take(MAX_NAME_LENGTH).collect();
+
+        let sanitized_src = self
+            .src
+            .trim()
+            .chars()
+            .take(MAX_SRC_LENGTH)
+            .collect::<String>();
+
+        let src = match Url::parse(&sanitized_src) {
+            Ok(_) => Some(sanitized_src),
+            Err(_) => None, // Invalid src URL, set to None
+        };
+
+        let content_type = self.content_type.trim().to_string();
+
+        Self {
+            name,
+            created_at: self.created_at,
```

Comment:
I don't understand why we cannot use `hash_id`. We have a `created_at` timestamp anyway, therefore it is duplicated information. It's a straightforward method for the user to post 3 times the same `.gif` without creating 3 times the file on his homeserver or the indexer.

Alternatively, we have `name` field, that could as well be the file and blob name instead.

Reply:
What is the emphasis on hash_id for? I understand its use case for something like tags, but we don't need that here.  
The use case you're mentioning is not solved by having the same hash_id.  With the current way, a sane user can create one blob, one file and then use that one file anywhere he wants.  Should we stop the ability of someone to create multiple blobs and multiple files? I don't think so. I think it's important to remember we're a proxy here and people can do whatever they want with their homeservers, so while we enable, we shouldn't really look into ways of prohibiting some use cases, no matter how fringe they might sound to us, when there's no problem with having them.

Reply:
Sorry, not necessarily saying `hash_id` should go on the `/file` object, I am talking about `hash_id` for the data blob. Are the `/blobs` and the way Nexus uses them covered by the specs?

There is no use for the timestamp ids.

Consider what happens when in pubky-app I am a shit poster and I reply to 10 users by dropping into the post editor modal the same animated gif to make fun of them.

Reply:
For blob ids it does make sense to have hash_ids.